frontend/src/services/auth-service.ts

```diff
@@ -0,0 +1,30 @@
+import { UserTokens } from 'shared-types';
+import apiClient from './api-client';
+
+export class AuthenticationService {
+  endpoint: string = '/auth';
+
+  async login(email: string, password: string, signal?: AbortSignal): Promise<UserTokens> {
+    return (
+      await apiClient.post<UserTokens>(
```

Comment:
maybe you can set the `this.endpoint` to be a `baseURL` or the apiClient object instead of prefix it

Reply:
Or you can have the api-client.ts file export a function which receives a `route` string and returns an axios instance configured to request to this path.
Ex: If i call
```ts
const authClient = createApiClient('/auth');
// authClient will be an AxiosInstance whose baseURL is configured to `http://localhost:8000/auth`
```

Reply:
This is super Nice To Have and not mandatory. feel free to merge without this change